text/0000-grid-primitives.md

```diff
@@ -0,0 +1,680 @@
+    lat_long_value=LatLong(
+        latitude=44977753,
+        longitude=-93265015)
+)
+```
+
+Due to the use of protobuf, the default values for `LatLong` would be `(0, 0)`.
+While this is a valid lat/long, it could be used to indicate an error, depending
+on the choice of the smart-contract implementer.
+
+## Schema Example
+
+A complete object representation can be built from the property definition
+messages, and instances can be represented by constructing items with the
+property value messages.  
+
+Suppose there is a requirement to store different types of light bulbs as part
+of an application. A lightbulb may consist of the properties size, bulb type,
+energy rating, and color.
+
+We could define a Lightbulb schema as follows:
+
+```
+Schema(
+    name="Lightbulb",
+    description="Example Lightbulb schema",
+    owner = "philips001"
+    properties=[
+        PropertyDefinition(
+            name="size",
+            data_type=PropertyDefinition.DataType.NUMBER,
+            description="Lightbulb radius, in millimeters",
+            number_exponent=0,
+            required=True
+        ),
+        PropertyDefinition(
+            name="bulb_type",
+            data_type=PropertyDefinition.DataType.ENUM,
+            enum_options=["filament", "CF", "LED"],
+            required=True
+        ),
+        PropertyDefinition(
+            name="energy_rating",
+            data_type=PropertyDefinition.DataType.NUMBER,
+            number_exponent=-2,
+            description="EnergyStar energy rating (percent)",
+        )
+        PropertyDefinition(
+            name="color",
+            data_type=PropertyDefinition.DataType.STRUCT,
+            description="A named RGB Color value",
+            struct_properties=[
+                PropertyDefinition(
+                    name='name',
+                    data_type=PropertyDefinition.DataType.STRING,
+                ),
+                PropertyDefinition(
+                    name='rgb_hex',
+                    data_type=PropertyDefinition.DataType.STRING,
+                )])])
+```
+
+Note: This example looks very similar to defining a struct property, but the
+fields in a schema may be optional.
+
+We can define a data structure that uses this schema for validation as follows:
+
+```
+message Lightbulb {
+    string id = 1;
+    string production_org = 2;
+    repeated PropertyValues properties = 3;
+} 
+```
+
+A Lightbulb smart contract would be responsible for validating the properties
+against the Lightbulb schema.
+
+## Addressing
+
+### Referencing Schemas
+
+Schemas are uniquely referenced by their unique schema name. For example:
+
+```
+    get_schema(schema_name)
+    set_schema(schema_name, schema)
+```
+
+### Schema Addressing in the Merkle-Radix State System
+
+All Grid addresses start with the 6-hex-character namespace prefix "621dee".
+Schemas are further prefixed under the Grid namespace with reserved "01", which
```

Comment:
"further prefixed" means that the 01 goes in front of 621dee, which is not the case.  I suggest rewriting to clarify that 01 follows the 621d22 prefix.